onmt/transforms/denoising.py

```diff
@@ -322,10 +322,11 @@ def __repr__(self):
         return '{}({})'.format(cls_name, cls_args)
 
 
-@register_transform(name='bart')
-class BARTNoiseTransform(Transform):
+@register_transform(name='ae_noise')
+class NoiseTransform(Transform):
     def __init__(self, opts):
         super().__init__(opts)
+        self.denoising_objective = opts.denoising_objective
```

Comment:
We should validate that `mask-random` is not used with the MASS objective.

In BART, it is possible to occasionally substitute a random token instead of the mask token. This slightly alleviates the tendency to copy all unmasked tokens verbatim.

In MASS, substituting a random token on the source side would also affect the target sequence: the random token would be complemented into a mask token, and would not contribute to the loss. This does not make sense.

Reply:
so `random_ratio > 0`